test/com/facebook/buck/go/GoAssumptions.java

```diff
@@ -64,4 +71,55 @@ public static void assumeGoCompilerAvailable() throws InterruptedException, IOEx
     }
     assumeNoException(exception);
   }
+
+  public static void assumeGoVersionAtLeast(String minimumVersion) {
+    List<Integer> minimumVersionNumbers = getVersionNumbers(minimumVersion);
+    Assume.assumeTrue(
+        "Expect minimum version string in the form of x.y.z, got" + minimumVersion,
+        minimumVersionNumbers.size() == 3);
+    Throwable exception = null;
+    String actualVersion = "0.0.0";
+    ProcessExecutor processExecutor = new DefaultProcessExecutor(new TestConsole());
+    try {
+      ProcessExecutor.Result goToolResult =
+          processExecutor.launchAndExecute(
+              ProcessExecutorParams.builder().addCommand("go", "version").build(),
+              EnumSet.of(ProcessExecutor.Option.EXPECTING_STD_OUT),
+              /* stdin */ Optional.empty(),
+              /* timeOutMs */ Optional.empty(),
+              /* timeoutHandler */ Optional.empty());
+      if (goToolResult.getExitCode() == 0) {
+        // e.g., "go version go1.9.2 darwin/amd64", "go version go1.9.2 windows/amd64",
+        // "go version go1.2.1 linux/amd64"
+        actualVersion = goToolResult.getStdout().get().substring(13, 18);
```

Comment:
That wouldn't work for versions like '1.10.123'. Could you use regex instead? Something like `((\d)+)\.((\d)+)\.((\d)+)` ?

Reply:
Good catch! Updated to use regex.